sdk/typespec/typespec_client_core/src/http/policies/logging.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use crate::http::{
+    policies::{Policy, PolicyResult},
+    sanitizer::{Sanitizer, DEFAULT_ALLOWED_HEADER_NAMES, DEFAULT_ALLOWED_QUERY_PARAMETERS},
+    Context, LoggingOptions, Request,
+};
+use std::collections::HashSet;
+use std::sync::Arc;
+use tracing::info;
+
+/// [`Policy`] to log  a request and response.
+#[derive(Clone, Debug, Default)]
+pub struct LoggingPolicy {
+    allowed_headers: HashSet<&'static str>,
```

Comment:
Do we think this will be large enough to warrant the overhead of a set? IIRC, most languages use an array. If we use a `Vec` we can order and use a `binary_search` which might be good enough.

Reply:
FWIW, Go uses a `map`, as does javascript, .Net uses a list, still working on Python.